File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = []
    if s[0] == ')' or s[-1] == '(':
        return False
    for i in s:
        if i == '(':
            answer.append(1)
        else:
            if len(answer) != 0:
                answer.pop()
            else:
                return False
    if len(answer) != 0:
        return False
    return True

File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.java
import java.util.Stack;

class Solution {
    public static boolean solution(String s) {
        Stack<Integer> stack = new Stack<>();

        for (char c : s.toCharArray()){
            if (c == '('){
                stack.add(0);
            }else{
                if (stack.isEmpty()){
                    return false;
                }else{
                    stack.pop();
                }
            }
        }
        return stack.isEmpty();
    }
}